### PR TITLE
applications: nrf_desktop: Add led stream to gaming mouse 

### PR DESCRIPTION
**Pull request by mip1-nordic**
applications: nrf_desktop: Add led stream to gaming mouse 
Create special led effect to serve incoming led stream.

Jira:DESK-431

Signed-off-by: Michal Patyk <michal.patyk@nordicsemi.no>

### CODE REVIEW

**Comment by mip1-nordic** [comment]
@MarekPieta

File: applications/nrf_desktop/configuration/nrf52840_pca20041/leds_def.h [hunk 1]
```diff
@@ -25,5 +25,10 @@ static const size_t led_pins[CONFIG_DESKTOP_LED_COUNT]
 		DT_NORDIC_NRF_PWM_PWM_1_CH0_PIN,
 		DT_NORDIC_NRF_PWM_PWM_1_CH1_PIN,
 		DT_NORDIC_NRF_PWM_PWM_1_CH2_PIN
-	}
+	},
+    {
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840_pca20041/leds_def.h]: indent

File: applications/nrf_desktop/configuration/common/led_effect.h [hunk 2]
```diff
@@ -31,9 +31,12 @@ struct led_effect_step {
 };
 
 struct led_effect {
-	const struct led_effect_step *steps;
+	struct led_effect_step *steps;
 	u16_t step_count;
 	bool loop_forever;
+	bool stream;
+    u16_t actual_step;
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/common/led_effect.h]: indent

**Comment by mip1-nordic** [comment]
Closed #4 .

**Pull request closed without merging**